test/e2e/alertmanager_test.go

```diff
@@ -231,3 +226,188 @@ receivers:
 		t.Fatal(err)
 	}
 }
+
+func TestAlertmanagerZeroDowntimeRollingDeployment(t *testing.T) {
+	t.Parallel()
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	ns := ctx.CreateNamespace(t, framework.KubeClient)
+	ctx.SetupPrometheusRBAC(t, ns, framework.KubeClient)
+
+	whReplicas := int32(1)
+	whdpl := &v1beta1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "alertmanager-webhook",
+		},
+		Spec: v1beta1.DeploymentSpec{
+			Replicas: &whReplicas,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "alertmanager-webhook",
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "webhook-server",
+							Image: "quay.io/brancz/prometheus-alertmanager-test-webhook",
```

Comment:
@brancz Can you move this to a CoreOS repo as suggested in your PR description. I think that would be easier in terms of managing permissions in case I need to push a new image.